File: MiniDOMFinanziaria/Finanziaria/Tabelle Spese/TabelleSpese.cs
using minidom.internals;

namespace minidom
{
    namespace internals
    {


        /// <summary>
    /// Classe Generica per l'accesso allae tabelle spese
    /// </summary>
    /// <remarks></remarks>
        public sealed class CTabelleSpeseClass : CModulesClass<Finanziaria.CTabellaSpese>
        {
            internal CTabelleSpeseClass() : base("modCQSPDTabelleSpese", typeof(Finanziaria.CTabellaSpeseCursor), -1)
            {
            }

            public CCollection<Finanziaria.CTabellaSpese> GetTabelleByCessionario(int cid, bool ov = true)
            {
                var ret = new CCollection<Finanziaria.CTabellaSpese>();
                foreach (Finanziaria.CTabellaSpese item in LoadAll())
                {
                    if (item.CessionarioID == cid && (ov == false || item.IsValid()))
                        ret.Add(item);
                }

                return ret;
            }
        }
    }

    public partial class Finanziaria
    {
        private static CTabelleSpeseClass m_TabelleSpese = null;

        public static CTabelleSpeseClass TabelleSpese
        {
            get
            {
                if (m_TabelleSpese is null)
                    m_TabelleSpese = new CTabelleSpeseClass();
                return m_TabelleSpese;
            }
        }
    }
}

File: MiniDOMFinanziaria/Finanziaria/Tabelle Spese/TabelleSpese.vb
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Finanziaria
Imports minidom.Internals

Namespace Internals


    ''' <summary>
    ''' Classe Generica per l'accesso allae tabelle spese
    ''' </summary>
    ''' <remarks></remarks>
    Public NotInheritable Class CTabelleSpeseClass
        Inherits CModulesClass(Of CTabellaSpese)

        Friend Sub New()
            MyBase.New("modCQSPDTabelleSpese", GetType(CTabellaSpeseCursor), -1)
        End Sub

        Public Function GetTabelleByCessionario(ByVal cid As Integer, Optional ByVal ov As Boolean = True) As CCollection(Of CTabellaSpese)
            Dim ret As New CCollection(Of CTabellaSpese)
            For Each item As CTabellaSpese In Me.LoadAll
                If (item.CessionarioID = cid AndAlso (ov = False OrElse item.IsValid)) Then ret.Add(item)
            Next
            Return ret
        End Function


    End Class
End Namespace

Partial Public Class Finanziaria




    Private Shared m_TabelleSpese As CTabelleSpeseClass = Nothing

    Public Shared ReadOnly Property TabelleSpese As CTabelleSpeseClass
        Get
            If (m_TabelleSpese Is Nothing) Then m_TabelleSpese = New CTabelleSpeseClass
            Return m_TabelleSpese
        End Get
    End Property
End Class
